### PR TITLE
test (JAVA SPRING RESTCLIENT) Add echo tests for the Spring 6 RestClient

### PR DESCRIPTION
**Pull request by Nicklas2751**
I added the Spring 6 RestClient to the Readme and implemented some of the echo tests for it.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@cachescrubber @welshm @MelleD @atextor @manedev79 @javisst @borsch @banlevente @Zomzog @martin-mfg

### CODE REVIEW

**Comment by wing328** [comment]
can you please add the new output folders to the workflow as well?

https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/samples-java-client-echo-api-jdk11.yaml
https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/samples-java-client-echo-api-jdk17.yaml
https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/samples-java-client-echo-api-jdk8.yaml

**Comment by Nicklas2751** [comment]
@wing328 Im sorry, but I don't understand what I should do/change. 😅

**Comment by wing328** [comment]
for example, in `https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/samples-java-client-echo-api-jdk8.yaml`, 

add the new folder path to line 10, 16 and 30

**Comment by Nicklas2751** [comment]
@wing328 I added the other clients to the JDK 17 tests but I can't add the Spring 6 RestClient tests to the JDK 8 or 11 echo api tests because the minimal version of the Spring RestClient is JDK 17.

File: modules/openapi-generator/src/main/resources/Java/libraries/restclient/ApiClient.mustache [hunk 1]
```diff
@@ -12,8 +12,8 @@ import org.springframework.http.HttpHeaders;
 import org.springframework.http.HttpMethod;
 import org.springframework.http.InvalidMediaTypeException;
 import org.springframework.http.MediaType;
-  import org.springframework.http.converter.HttpMessageConverter;
-  import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/Java/libraries/restclient/ApiClient.mustache:16]: thanks for improving the code format

**Comment by martin-mfg** [comment]
Hi @Nicklas2751 and @wing328, it looks like this PR did not make any changes in https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/ , even though you discussed this. Is this intended?

**Comment by Nicklas2751** [comment]
> Hi @Nicklas2751 and @wing328, it looks like this PR did not make any changes in https://github.com/OpenAPITools/openapi-generator/blob/master/.github/workflows/ , even though you discussed this. Is this intended?

Oh yes you are right there is a commit missing. Thanks for watching out.

**Pull request merged**